integration-tests/tests/src/node/path.ts

```diff
@@ -36,24 +38,52 @@ const schema = {
   },
 };
 
+const FlexibleSchema = { ...Schema, properties: { ...Schema.properties, nonQueryable: "string?" } };
+
 describe("path configuration (local)", function () {
   it("relative path", function () {
     const filename = getRelativePath();
-    const realm = new Realm({ path: filename, schema: [schema] });
+    const realm = new Realm({ path: filename, schema: [Schema] });
     expect(realm.path.endsWith(filename)).to.be.true;
     realm.close();
     Realm.deleteFile({ path: filename });
   });
 
   it("absolute path", function () {
     const filename = getAbsolutePath();
-    const realm = new Realm({ path: filename, schema: [schema] });
+    const realm = new Realm({ path: filename, schema: [Schema] });
     expect(realm.path).to.equal(filename);
     realm.close();
     Realm.deleteFile({ path: filename });
   });
 });
 
+describe.skipIf(environment.missingServer, `app configuration of root directory (flexible sync)`, async function () {
+  const { appId, baseUrl } = await importApp("with-db-flx");
+
+  it.only("directory and file created where expected", async function () {
+    const tmpdir = getAbsolutePath();
+    expect(fs.exists(tmpdir)).to.be.false;
+
+    const app = new Realm.App({ id: appId, baseUrl, syncRootDirectory: tmpdir });
```

Comment:
I would've expected `syncRootDirectory` to be set on the `Realm`.  Is that how other sdks are doing it as well? 

 Also, what would happen when the `path` is set on the `Realm` to something outside of the `syncRootDirectory`?

Reply:
> I would've expected syncRootDirectory to be set on the Realm. Is that how other sdks are doing it as well?

It is a configuration parameter for the sync client which is initialised when calling the `Realm.App` constructor. If on the Realm configuration, it will be too late. Moreover, the metadata directory is _per_ app and not _per_ Realm.

* [DotNet - baseFilePath](https://www.mongodb.com/docs/realm-sdks/dotnet/latest/reference/Realms.Sync.AppConfiguration.html#Realms_Sync_AppConfiguration_BaseFilePath)
* [Java - syncRootDirectory](https://www.mongodb.com/docs/realm-sdks/java/latest/io/realm/mongodb/AppConfiguration.Builder.html#syncRootDirectory(java.io.File))

> Also, what would happen when the path is set

For absolute path, the Realm file is stored outside the sync root directory and metadata within. For relative path, both metadata and Realm file will be stored relative to the sync root directory.

Reply:
Makes sense!  Thanks for the clarification